File: ktor/kotlin/src/main/kotlin/EchoData.kt
import kotlinx.serialization.Serializable

@Serializable
data class EchoData(
    val id: Int,
    val message: String,
    val innerDataList: List<InnerData>
)

@Serializable
data class InnerData(
    val id: Int,
    val number: Double,
    val itemList: List<ListItem>
)

@Serializable
data class ListItem(
    val id: Int,
    val boolean: Boolean,
    val string: String,
)

File: vapor/swift/Sources/App/EchoData.swift
import Foundation
import Vapor

struct EchoData: Codable, Content {
    let id: Int
    let message: String
    let innerDataList: [InnerData]
}

struct InnerData: Codable, Content {
    let id: Int
    let number: Double
    let itemList: [ListItem]
}

struct ListItem: Codable, Content {
    let id: Int
    let boolean: Bool
    let string: String
}
